vllm/v1/spec_decode/eagle.py

```diff
@@ -1003,52 +1004,96 @@ def load_model(self, target_model: nn.Module) -> None:
                     "Target model does not have 'embed_tokens' or 'embedding' attribute"
                 )
 
-            # Check if shapes match and we found the embedding
-            eagle_shape = self.model.model.embed_tokens.weight.shape
-            target_shape = target_embed_tokens.weight.shape
-            if eagle_shape == target_shape:
-                logger.info(
-                    "Assuming the EAGLE head shares the same vocab embedding"
-                    " with the target model."
-                )
-                del self.model.model.embed_tokens
-                self.model.model.embed_tokens = target_embed_tokens
+            share_embeddings = False
+            reason = ""
+            if hasattr(self.model, "has_own_embed_tokens"):
+                # EAGLE-{1,2,3} model
+                if not self.model.has_own_embed_tokens:
+                    share_embeddings = True
+                    reason = (
```

Comment:
Why have "reason" if it's just going to get passed into `logger.info` anyways? Can't you just call it here instead of having a variable?